app/uk/gov/hmrc/tariffclassificationfrontend/controllers/AssignCaseController.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.controllers
+
+import javax.inject.{Inject, Singleton}
+import play.api.i18n.MessagesApi
+import play.api.mvc._
+import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+import uk.gov.hmrc.tariffclassificationfrontend.models.Case
+import uk.gov.hmrc.tariffclassificationfrontend.models.request.AuthenticatedRequest
+import uk.gov.hmrc.tariffclassificationfrontend.service.CasesService
+import uk.gov.hmrc.tariffclassificationfrontend.views
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.Future.successful
+
+@Singleton
+class AssignCaseController @Inject()(authenticatedAction: AuthenticatedAction,
+                                     override val caseService: CasesService,
+                                     val messagesApi: MessagesApi,
+                                     override implicit val config: AppConfig) extends RenderCaseAction {
+
+
+  override protected def redirect: String => Call = routes.CaseController.trader
+
+  override protected def isValidCase(c: Case)(implicit request: AuthenticatedRequest[_]): Boolean = c.queueId.isDefined && !c.assignee.map(_.id).contains(request.operator.id)
```

Comment:
I know `Option.contains(x)` means  `== Some(x)`, but I find the `contains` confusing, especially if you are used to the `String.contains`.

How about rewriting the conditions with `==` or some other way?

Reply:
I think you want to express the fact that the case is not assigned to anyone, or assigned to another user.
Right?

You could say:

```
override protected def isValidCase(c: Case)(implicit request: AuthenticatedRequest[_]): Boolean = {
    (c.queueId, c.assignee) match {
       case (Some(_), Some(id)) if id != request.operator.id => true
       case _ => false
    }
}
```

Reply:
That seems more readable, i'll change it

Reply:
Pushed